crates/matrix-sdk-crypto/src/olm/signing/pk_signing.rs

```diff
@@ -311,27 +313,18 @@ impl Signing {
         CrossSigningKey::new(user_id, vec![usage], keys, BTreeMap::new())
     }
 
+    pub fn sign(&self, message: &str) -> Ed25519Signature {
+        self.inner.sign(message.as_bytes())
+    }
+
     #[cfg(test)]
-    pub fn verify(
+    pub fn verify_json(
         &self,
-        message: &str,
-        signature: &Ed25519Signature,
+        user_id: &ruma::UserId,
+        key_id: &DeviceKeyId,
+        message: &mut Value,
```

Comment:
Why exactly does this require a mutable reference to the message to verify it? Isn't verification a read-only action?

(This is just an oddity I'm picking up as i'm scrolling through this.)